server/etcdserver/server.go

```diff
@@ -1081,9 +1084,10 @@ func (s *EtcdServer) applyEntries(ep *etcdProgress, apply *apply) {
 }
 
 func (s *EtcdServer) triggerSnapshot(ep *etcdProgress) {
-	if ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
+	if !s.shouldSnapshot && ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
```

Comment:
This change may introduce additional readability complex. When other developers read this source code in future, they may need sometime to figure out when will the `shouldSnapshot` be `true` or `false`. 

I suggest to keep all all the reasons to create a snapshot in one place, and change the if to 
```
if !s.shouldSnapshot {
    return
}
```

Reply:
Agree that `shouldSnapshot` could be confusing, however I don't agree that we should unify it under variable. I think checking both variable and etcdProgress makes more sense.  Renamed to `forceSnapshot`.